diff --git a/python/huggingfaceserver/huggingfaceserver/vllm/vllm_model.py b/python/huggingfaceserver/huggingfaceserver/vllm/vllm_model.py
index 8206f88d2..7f9c07558 100644
--- a/python/huggingfaceserver/huggingfaceserver/vllm/vllm_model.py
+++ b/python/huggingfaceserver/huggingfaceserver/vllm/vllm_model.py
@@ -29,7 +29,7 @@ from vllm.tool_parsers import ToolParserManager
 from vllm.entrypoints.openai.models.protocol import BaseModelPath
 from vllm.entrypoints.openai.models.serving import OpenAIServingModels
 from vllm.entrypoints.openai.cli_args import validate_parsed_serve_args
-from vllm.entrypoints.chat_utils import load_chat_template
+from vllm.entrypoints.chat_utils import load_chat_template, ChatTemplateConfig
 from vllm.entrypoints.openai.engine.protocol import ErrorResponse as engineError
 from vllm.reasoning import ReasoningParserManager
 
@@ -196,9 +196,11 @@ class VLLMModel(OpenAIEncoderModel, OpenAIGenerativeModel):  # pylint:disable=c-
                     self.engine_client,
                     self.openai_serving_models,
                     request_logger=self.request_logger,
-                    chat_template=resolved_chat_template,
-                    chat_template_content_format=self.args.chat_template_content_format,
-                    trust_request_chat_template=self.args.trust_request_chat_template,
+                    chat_template_config=ChatTemplateConfig(
+                        chat_template=resolved_chat_template,
+                        chat_template_content_format=self.args.chat_template_content_format,
+                        trust_request_chat_template=self.args.trust_request_chat_template,
+                    ),
                     log_error_stack=self.args.log_error_stack,
                 )
                 if "embed" in supported_tasks
@@ -210,6 +212,12 @@ class VLLMModel(OpenAIEncoderModel, OpenAIGenerativeModel):  # pylint:disable=c-
                     self.engine_client,
                     self.openai_serving_models,
                     request_logger=self.request_logger,
+                    supported_tasks=supported_tasks,
+                    chat_template_config=ChatTemplateConfig(
+                        chat_template=resolved_chat_template,
+                        chat_template_content_format=self.args.chat_template_content_format,
+                        trust_request_chat_template=self.args.trust_request_chat_template,
+                    ),
                     log_error_stack=self.args.log_error_stack,
                 )
                 if ("embed" in supported_tasks or "classify" in supported_tasks)
